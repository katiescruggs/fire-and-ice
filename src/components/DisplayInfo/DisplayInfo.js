import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchHouses } from '../../helpers/apiCalls';
import { setHouses } from '../../actions';
import CardContainer from '../CardContainer/CardContainer';
import PropTypes, { shape, func, string } from 'prop-types';

export class DisplayInfo extends Component {
  async componentDidMount() {
    const houses = await fetchHouses();
    this.props.setHouses(houses);
  }

  render() {
    return (
      <div className='Display-info'>
            
          {this.props.houses.length === 0 && 
            <img src='../../wolf.gif' />
          }

          {this.props.houses.length > 0 &&
            <CardContainer />
          }
          
      </div>
      );
  }
}

export const mapStateToProps = ({ houses }) => ({ houses });
export const mapDispatchToProps = dispatch => ({ 
  setHouses: (houses) => dispatch(setHouses(houses))
});

DisplayInfo.propTypes = {

};

export default connect(mapStateToProps, mapDispatchToProps)(DisplayInfo);