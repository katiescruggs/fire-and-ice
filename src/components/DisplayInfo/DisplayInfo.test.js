import React from 'react';
import { shallow } from 'enzyme';
import { DisplayInfo, mapStateToProps, mapDispatchToProps} from './DisplayInfo';

describe('DisplayInfo Container', () => {
  describe('DisplayInfo', () => {
    let displayInfo;
    const mockHouses = ['an', 'array', 'of', 'houses'];
    const mockSetHouses = jest.fn();

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        houses: ['array', 'of', 'houses']
      })
    }));

    beforeEach(() => {
      displayInfo = shallow(<DisplayInfo houses={mockHouses} setHouses={mockSetHouses} />);
    });

    it('should exist', () => {
      expect(displayInfo).toBeDefined();
    });

    it('should match snapshot', () => {
      expect(displayInfo).toMatchSnapshot();
    });

    it('should display connected card container if there are houses', () => {
      displayInfo = shallow(<DisplayInfo houses={mockHouses} setHouses={mockSetHouses} />);
      expect(displayInfo.find('Connect').length).toEqual(1);
      expect(displayInfo.find('img').length).toEqual(0);
    });

    it('should display loading gif if there are no houses', () => {
      displayInfo = shallow(<DisplayInfo houses={[]} setHouses={mockSetHouses} />);

      expect(displayInfo.find('img').length).toEqual(1);
      expect(displayInfo.find('Connect').length).toEqual(0);
    });
  });

  describe('mapStateToProps', () => {
    it('connects houses to props', () => {
      const mockStore = {
        houses: ['array', 'of', 'houses']
      };

      const result = mapStateToProps(mockStore);

      expect(result.houses).toEqual(mockStore.houses);
    });
  });

  describe('mapDispatchToProps', () => {
    it('connects dispatch to setHouses in props', () => {
      const mockDispatch = jest.fn();
      const result = mapDispatchToProps(mockDispatch);

      result.setHouses();

      expect(mockDispatch).toHaveBeenCalled;
    });
  });
});