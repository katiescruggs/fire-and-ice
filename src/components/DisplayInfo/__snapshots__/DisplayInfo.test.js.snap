// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayInfo Container DisplayInfo should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayInfo
    houses={
        Array [
            "an",
            "array",
            "of",
            "houses",
          ]
    }
    setHouses={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <Connect(CardContainer) />,
      ],
      "className": "Display-info",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <Connect(CardContainer) />,
        ],
        "className": "Display-info",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
